File: go/347_Top_K_Frequent_Elements/347_Top_K_Frequent_Elements.go
func topKFrequent(nums []int, k int) []int {
    m := make(map[int]int, len(nums))
    for _, n := range nums {
        m[n] += 1
    }

    keyList := make([]int, len(m))

    i := 0
    for k := range m {
        keyList[i] = k
        i++
    }

    sort.Slice(keyList, func(i int, j int) bool {
        return m[keyList[i]] > m[keyList[j]]
    })

    return keyList[:k]
}

File: cpp/347_Top_K_Frequent_Elements/347_Top_K_Frequent_Elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int, int> m;
        for (int n : nums) {
            m[n] += 1;
        }
        
        vector<pair<int, int>> sortedNumbers;
        for (auto key : m) {
            sortedNumbers.push_back({key.second, key.first});
        }
        sort(sortedNumbers.rbegin(), sortedNumbers.rend());
        
        vector<int> result;
        for (int i = 0; i < k; i++) {
            result.push_back(sortedNumbers[i].second);
        }
        
        return result;
    }
};
